### PR TITLE
fix: 4.x gif thumb url base64 type should be corrected

### PR DESCRIPTION
**Pull request by linxianxi**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/pull/44083  这个 pr 修复到 4.x
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix the bug that Upload gif thumbnail will not play      |
| 🇨🇳 Chinese |     修复 Upload gif 缩略图不会动的问题      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 55d5318</samp>

This pull request fixes some issues and adds some features related to the upload component. It prevents errors and memory leaks from object URLs in tests, adds a test case for gif preview, and improves the `fileToObject` function to handle gif files and release object URLs.

### <samp>🤖 Generated by Copilot at 55d5318</samp>

*  Add and restore mock for `window.URL.revokeObjectURL` in tests to avoid jsdom errors ([link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390L32-R39), [link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390R93))
*  Add parentheses around arrow function parameters for consistency and readability ([link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390L681-R684), [link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390L804-R812), [link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390L889-R892), [link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390L918-R921), [link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390L942-R970), [link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390L1048-R1076), [link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390L1142-R1170), [link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390L1231-R1259), [link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390L1247-R1275), [link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-dcf89e801a7290d050d09b02555f937ff87477b6df8810c84d8193ac2aaf67f0L31-R36), [link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-dcf89e801a7290d050d09b02555f937ff87477b6df8810c84d8193ac2aaf67f0L78-R78))
*  Release object URL after converting canvas to data URL in `fileToObjectUrl` to avoid memory leaks ([link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-dcf89e801a7290d050d09b02555f937ff87477b6df8810c84d8193ac2aaf67f0L110-R110))
*  Handle gif file type in `fileToObjectUrl` by reading data URL directly without canvas ([link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-dcf89e801a7290d050d09b02555f937ff87477b6df8810c84d8193ac2aaf67f0L116-R125))
*  Add test case for uploading gif file and previewing in base64 format ([link](https://github.com/ant-design/ant-design/pull/44129/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390R927-R951))

### CODE REVIEW

File: components/date-picker/__tests__/__snapshots__/other.test.tsx.snap [hunk 1]
```diff
@@ -303,7 +303,7 @@ exports[`MonthPicker and WeekPicker render WeekPicker 1`] = `
               </thead>
               <tbody>
                 <tr
-                  class="ant-picker-week-panel-row ant-picker-week-panel-row-selected"
+                  class="ant-picker-week-panel-row"
```
Comment (afc163) [inline components/date-picker/__tests__/__snapshots__/other.test.tsx.snap:306]: 为啥这个会变？

Comment (linxianxi) [reply components/date-picker/__tests__/__snapshots__/other.test.tsx.snap:306]: https://github.com/react-component/picker/pull/664 貌似是这个，修复了 4.x 的

Comment (afc163) [reply components/date-picker/__tests__/__snapshots__/other.test.tsx.snap:306]: @zombieJ 看看是否是预期行为。

Comment (zombieJ) [reply components/date-picker/__tests__/__snapshots__/other.test.tsx.snap:306]: 嗯，预期的。rc-picker patch 也是从 v5 里 pick 的

**Review by zombieJ — APPROVED** [review]

**Pull request merged**